test/modules/landing/landing_router_test.dart

```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:quick_test/quick_test.dart';
+import 'package:survey/modules/home/home_module.dart';
+import 'package:survey/modules/landing/landing_module.dart';
+import 'package:survey/modules/login/login_module.dart';
+
+import '../../mocks/build_context.dart';
+import '../../mocks/navigator_state.dart';
+
+class FakeAnimation extends Mock implements Animation<double> {}
+
+void main() {
+  describe("a Landing router", () {
+    late LandingRouter router;
+    late MockBuildContext buildContext;
+    late MockNavigatorState navigatorState;
+
+    beforeEach(() {
+      buildContext = MockBuildContext();
+      navigatorState = MockNavigatorState();
+
+      router = LandingRouterImpl();
+    });
+
+    describe("it's replaceToLoginScreen() is called", () {
+      beforeEach(() {
+        when(buildContext.findAncestorStateOfType<NavigatorState>())
+            .thenReturn(navigatorState);
+        when(navigatorState.pushReplacement(any))
+            .thenAnswer((_) => Future.value());
+        router.replaceToLoginScreen(context: buildContext);
+      });
+
+      it("triggers navigator to push replacement to Login screen", () {
+        final PageRouteBuilder builder =
+            verify(navigatorState.pushReplacement(captureAny)).captured.single
+                as PageRouteBuilder;
+        final widget =
+            builder.pageBuilder(buildContext, FakeAnimation(), FakeAnimation());
+        expect(widget, isA<LoginModule>());
+      });
+    });
```

Comment:
I think we should write `describe` like `describe("when replaceToLoginScreen() is called", ...)`, then `it("triggers navigator to push replacement to Login screen",...)`
`it's replaceToLoginScreen() is called` is meaningless imo

Reply:
@manh-t I follows their convention [here](https://github.com/Quick/Quick/blob/main/Documentation/en-us/QuickExamplesAndGroups.md#example-groups-using-describe-and-context)
So `when` will lead to a `context` rather than a `describe` 🤔

Reply:
@markgravity so at least we should update `describe` to be correct grammar 🙋 `it's replaceToLoginScreen() is called` -> `its replaceToLoginScreen is called`

Reply:
@manh-t oops! my bad. I fixed them!! 😁